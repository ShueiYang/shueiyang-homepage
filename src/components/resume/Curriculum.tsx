import { RESUME_DATA } from "@/constants/resume.data";
import { Fragment } from "react";

const Curriculum = () => {
  // use index as key since this data is static.
  return (
    <div className="grid grid-cols-4 gap-5">
      {RESUME_DATA.map((resume, index) => {
        return (
          <Fragment key={index}>
            <div className="col-span-1">{resume.year}</div>
            <div className="col-span-3">{resume.cv}</div>
          </Fragment>
        );
      })}
    </div>
  );
};

export default Curriculum;
