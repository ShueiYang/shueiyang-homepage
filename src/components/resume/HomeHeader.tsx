"use client";

import Image from "next/image";
import { useContext } from "react";
import { ThemeContext } from "@/providers/ThemeProvider";
import HeroTitle from "@/components/resume/HeroTitle";
import HeroName from "./HeroName";

export default function HomeHeader() {
  const { theme } = useContext(ThemeContext);

  return (
    <div className="mx-auto mb-3 flex flex-col items-center justify-center gap-[5%] sm:max-w-[95%] md:max-w-[80%] md:flex-row lg:max-w-2xl">
      <div
        className={`${
          theme === "light"
            ? "bg-gradient-to-r from-sand-l to-[#ece6b3]"
            : "from-[#30375e] to-sea-d dark:bg-gradient-to-r"
        } mt-5 grow rounded-lg p-5 text-center md:px-7 md:text-left`}
      >
        <h1 className="font-ibm text-2xl font-semibold tracking-wide sm:text-3xl lg:text-4xl">
          <HeroTitle />
          <HeroName />
          <p className="md: mt-2 font-robo text-lg font-medium sm:text-xl lg:text-[1.3rem]">
            Un développeur passionné par l&apos;apprentissage et la création
            dans le domaine du web.
          </p>
        </h1>
      </div>

      <div className="mt-4 shrink-0 sm:mt-0 ">
        <Image
          src="/images/kim.png"
          className="mt-6 inline-block rounded-full border-2 border-solid border-neutral-300 md:mt-4"
          width={100}
          height={100}
          alt="profile Image"
          priority
        />
      </div>
    </div>
  );
}
