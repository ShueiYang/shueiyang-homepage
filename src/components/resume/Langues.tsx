
const Langues = () => {
  
  return (
    <>
      <h2 className="section"> 
        Langues
      </h2>
      <div className="md:flex gap-14">
        <div className="flex gap-4">  
          <div>Chinois :</div>
          Mandarin langue maternelle
        </div>
        <div className="flex gap-4">  
          <div>Anglais :</div>
          Courant
        </div>
      </div>
    </>
  )
}

export default Langues;