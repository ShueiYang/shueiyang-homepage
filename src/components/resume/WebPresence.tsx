import SocialButton from "@/components/customButton/SocialButton";
import { IoLogoGithub, IoLogoLinkedin, IoLogoDiscord, IoLogoStackoverflow } from "react-icons/io5"
import { FaSteam,  FaBattleNet } from "react-icons/fa"
import ModalDialog from "@/components/modal/ModalDialog";

const WebPresence = () => {

  return (
    <>
      <h2 className="section sm:mb-5"> 
        Sur le Web
      </h2>
      <div className="sm:grid grid-cols-2 lg:grid-cols-4">
        <SocialButton iconName={IoLogoGithub} path="https://github.com/ShueiYang">
          @ShueiYang
        </SocialButton>
        <SocialButton iconName={IoLogoLinkedin} path="https://www.linkedin.com/in/shueiyang">
          @Shueiyang
        </SocialButton>
        <SocialButton iconName={FaSteam} path="https://steamcommunity.com/id/shueiyang">
          @Yangxuzhu
        </SocialButton>
        <SocialButton iconName={IoLogoDiscord} path="https://discord.com/users/282279109273583616">
          @Yang
        </SocialButton>
        <SocialButton iconName={IoLogoStackoverflow} path="https://stackoverflow.com/users/19114911/shueiyang">
          @ShueiYang
        </SocialButton>
        <div>
          <ModalDialog title="#Yang" />
        </div>
        <div>
          <ModalDialog title="@Yang" />
        </div>      
        <SocialButton iconName={FaBattleNet} path="https://account.battle.net">
          #YangWenXue
        </SocialButton>
      </div>
    </>
  )
}

export default WebPresence;