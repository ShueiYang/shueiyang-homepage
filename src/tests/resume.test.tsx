import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import HomeHeader from "@/components/resume/HomeHeader";
import Curriculum from "@/components/resume/Curriculum";
import Langues from "@/components/resume/Langues";
import WebPresence from "@/components/resume/WebPresence";

describe("Test components in resume", () => {
  test("Homeheader", async () => {
    render(<HomeHeader />);

    expect(
      screen.getByRole("heading", {
        name: /h e l l o , j e s u i s k i m un développeur passionné par l'apprentissage et la création dans le domaine du web\./i,
      }),
    ).toBeInTheDocument();
  });

  test("Curriculum", async () => {
    const { container } = render(<Curriculum />);
    expect(container).toMatchSnapshot();
  });

  test("Webpresence", async () => {
    render(<WebPresence />);

    expect(screen.getAllByRole("link", { name: /@shueiyang/i })).toHaveLength(
      3,
    );
    expect(screen.getAllByRole("button", { name: /@yang/i })).toHaveLength(3);
    expect(
      screen.getByRole("link", {
        name: /@yangxuzhu/i,
      }),
    ).toBeInTheDocument();
  });

  test("Langues", async () => {
    render(<Langues />);

    expect(
      screen.getByRole("heading", {
        name: /langues/i,
      }),
    ).toBeInTheDocument();

    expect(screen.getByText(/chinois :/i)).toBeInTheDocument();
    expect(screen.getByText(/mandarin langue maternelle/i)).toBeInTheDocument();
    expect(screen.getByText(/anglais :/i)).toBeInTheDocument();
    expect(screen.getByText(/courant/i)).toBeInTheDocument();
  });
});
