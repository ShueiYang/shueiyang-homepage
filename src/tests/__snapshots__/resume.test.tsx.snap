// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test components in resume > Curriculum 1`] = `
<div>
  <dl
    class="grid grid-cols-4 gap-5"
  >
    <dt
      class="col-span-1"
    >
      1983
    </dt>
    <dd
      class="col-span-3"
    >
      Né à Paris, France
    </dd>
    <dt
      class="col-span-1"
    >
      2002
    </dt>
    <dd
      class="col-span-3"
    >
      Bac Scientifique validé au Lycée Colbert
    </dd>
    <dt
      class="col-span-1"
    >
      2004 - 2008
    </dt>
    <dd
      class="col-span-3"
    >
      Vente et Assistant commercial à ISIS Maroquinerie (Osny 95)
    </dd>
    <dt
      class="col-span-1"
    >
      2009
    </dt>
    <dd
      class="col-span-3"
    >
      BTS en maintenance informatique et réseaux (ITIC Paris)
    </dd>
    <dt
      class="col-span-1"
    >
      2010 - 2022
    </dt>
    <dd
      class="col-span-3"
    >
      POBIX.COM: Vente et expédition des matériels informatiques 
            spécialisé en batteries et chargeurs pour pc portable, Mac, ainsi que
            les EPDM (Vélo Trotinette)
    </dd>
    <dt
      class="col-span-1"
    >
      Juin - Nov 2022
    </dt>
    <dd
      class="col-span-3"
    >
      Autoformation en Développment web et application (Validation de la certification
            Udemy "The Complete Web Developer" de Zero to Mastery en Août, suivi après 
            la certification Udemy "Complete NodeJs Developer" de ZTM en Novembre)
    </dd>
    <dt
      class="col-span-1"
    >
      Avril - Juin 2023
    </dt>
    <dd
      class="col-span-3"
    >
      Validation de la Certification du Reacteur (full stack JavaScript) et
            pré-qualification au passage du titre RNCP niveau 6 (Bac +3/4) de Concepteur
            développeur d'applications. (Le Reacteur à Paris)
    </dd>
    <dt
      class="col-span-1"
    >
      Sept - Fev 2024
    </dt>
    <dd
      class="col-span-3"
    >
      Stage de 6 mois au poste de Developpeur Front junior chez Ekino.
    </dd>
  </dl>
</div>
`;
